libglusterfs/src/iobuf.c

```diff
@@ -161,12 +163,26 @@ __iobuf_arena_alloc(struct iobuf_pool *iobuf_pool, size_t page_size,
 
     iobuf_arena->arena_size = rounded_size * num_iobufs;
 
-    iobuf_arena->mem_base = mmap(NULL, iobuf_arena->arena_size,
-                                 PROT_READ | PROT_WRITE,
-                                 MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+    iobuf_arena->mem_base = MAP_FAILED;
+
+#ifdef GF_LINUX_HOST_OS
+    /* On Linux, try to map with 2MB huge pages first */
+    iobuf_arena->mem_base = mmap(
```

Comment:
By default mmap failed when call with MAP_HUGETLB flag because nr_hugepages is 0, it will not work unless user does not set the nr_hugepages.Do we need to log the same in message and set the nr_hugepages also?
Do u have any idea about improvement ?

Reply:
I leave it up to to the user to configure, as we also can't estimate the amount needed.
As for performance, I'd be happy to see some performance numbers on CI.

I assume to see the impact we'll also need to assess if we want 128K to come from the small, regular tcmalloc allocation or the mmap based.

Reply:
sure, i will do.

Reply:
To test, as root:
```
echo 20 > /proc/sys/vm/nr_hugepages
```
should do the trick.
And you can see the impact:
```
[ykaul@ykaul glusterfs]$ fgrep  -i huge /proc/meminfo  
AnonHugePages:         0 kB
ShmemHugePages:        0 kB
FileHugePages:         0 kB
HugePages_Total:      20    <---
HugePages_Free:       18   <---
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
Hugetlb:           40960 kB
```